src/compiler/checker.ts

```diff
@@ -344,7 +344,8 @@ namespace ts {
         });
         const typeofType = createTypeofType();
 
-        let jsxElementType: Type;
+        let getJsxElementType: () => Type;

```

Comment:
I change this to use memoize so it is similar to how we handle other type (see below)....now to think about it,  it may have additional cause of function call...